File: backend/models/conversation/conversation.py
from enum import Enum, unique
from typing import Dict, List

from models.tts.viseme import Viseme, WordOffset
from pydantic import BaseModel


@unique
class ConversationMessageType(str, Enum):
    QUERY: str = "query"
    AUDIO_RESPONSE: str = "audio_response"


class ConversationMessage(BaseModel):
    type: ConversationMessageType
    data: BaseModel

class QueryMessage(BaseModel):
    query: Dict[str, float]

class AudioMessage(BaseModel):
    base64_audio: str
    viseme: List[Viseme]
    word_boundary: List[WordOffset]

File: frontend/src/types/avatar/conversation.ts
export enum ConversationMessageType {
  QUERY = "query",
  AUDIO_RESPONSE = "audio_response",
}

export interface ConversationMessage {
  type: ConversationMessageType;
  data: any;
}

export interface Viseme {
  stopTime: number;
  readyPlayerMeViseme: string;
}

export interface WordOffset {
  offset_duration: number;
  text_offset: number;
  word_length: number;
}

export interface AudioMessage {
  base64_audio: string;
  viseme: Viseme[];
  word_boundary: WordOffset[];
}
